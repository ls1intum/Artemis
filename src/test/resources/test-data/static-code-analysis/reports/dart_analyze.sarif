{
  "version": "2.1.0",
  "$schema": "https://docs.oasis-open.org/sarif/sarif/v2.1.0/errata01/os/schemas/sarif-schema-2.1.0.json",
  "runs": [
    {
      "tool": {
        "driver": {
          "name": "dart analyze",
          "version": "1",
          "informationUri": "https://dart.dev/tools/dart-analyze",
          "rules": [
            {
              "id": "empty_statements",
              "helpUri": "https://dart.dev/diagnostics/empty_statements",
              "properties": {
                "type": "LINT"
              }
            }
          ]
        }
      },
      "conversion": {
        "tool": {
          "driver": {
            "name": "analyze_sarif",
            "semanticVersion": "0.0.1",
            "informationUri": "https://github.com/ls1intum/dart_analyze_sarif"
          }
        }
      },
      "originalUriBaseIds": {
        "SRCROOT": {
          "uri": "file:///home/user/bust/"
        }
      },
      "results": [
        {
          "ruleId": "empty_statements",
          "ruleIndex": 0,
          "level": "note",
          "message": {
            "text": "Unnecessary empty statement."
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "lib/bust.dart",
                  "uriBaseId": "SRCROOT"
                },
                "region": {
                  "startLine": 2,
                  "startColumn": 3,
                  "endLine": 2,
                  "endColumn": 4,
                  "charOffset": 20,
                  "charLength": 1
                }
              }
            }
          ]
        }
      ]
    }
  ]
}
