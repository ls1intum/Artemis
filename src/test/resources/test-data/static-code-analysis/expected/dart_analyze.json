{
  "tool": "DART_ANALYZE",
  "issues": [
    {
      "filePath": "lib/bust.dart",
      "startLine": 2,
      "endLine": 2,
      "startColumn": 3,
      "endColumn": 4,
      "rule": "empty_statements",
      "category": "LINT",
      "message": "Unnecessary empty statement.",
      "priority": "note"
    }
  ]
}
